src/spiral_stc.cpp

```diff
@@ -8,294 +8,356 @@
+      float tool_radius_default = 0.5f;
+      declare_parameter_if_not_declared(node_, name_ + ".tool_radius", rclcpp::ParameterValue(tool_radius_default));
+      node_->get_parameter(name_ + ".tool_radius", tool_radius_);
+      initialized_ = true;
     }
-    else
+  }
+
+  void SpiralSTC::activate()
+  {
+    RCLCPP_INFO(
+      rclcpp::get_logger("FullCoveragePathPlanner"), "Activating plugin %s of type FullCoveragePathPlanner",
+      name_.c_str());
+  }
+
+  void SpiralSTC::deactivate()
+  {
+    RCLCPP_INFO(
+      rclcpp::get_logger("FullCoveragePathPlanner"), "Deactivating plugin %s of type FullCoveragePathPlanner",
+      name_.c_str());
+  }
+
+  void SpiralSTC::cleanup()
+  {
+    RCLCPP_INFO(
+      rclcpp::get_logger("FullCoveragePathPlanner"), "Cleaning up plugin %s of type FullCoveragePathPlanner",
+      name_.c_str());
+    // TODO(clopez) Add proper cleanup
+  }
+
+  nav_msgs::msg::Path SpiralSTC::createPlan(
+    const geometry_msgs::msg::PoseStamped & start,
+    const geometry_msgs::msg::PoseStamped & goal)
+  {
+    nav_msgs::msg::Path global_path;
+    std::vector<geometry_msgs::msg::PoseStamped> plan;
+
+    SpiralSTC::makePlan(start, goal, plan);
+
+    global_path.poses.clear();
+    global_path.header.stamp = node_->now();
+    global_path.header.frame_id = global_frame_;
+    for (auto pose : plan)
     {
-      // Initialize spiral direction towards y-axis
-      dx = 0;
-      dy = 1;
+      global_path.poses.push_back(pose);
     }
-    done = true;
-
-    for (int i = 0; i < 4; ++i)
+    return global_path;
+  }
+  std::list<gridNode_t> SpiralSTC::spiral(std::vector<std::vector<bool>> const &grid, std::list<gridNode_t> &init,
+                                          std::vector<std::vector<bool>> &visited)
+  {
+    int dx, dy, dx_prev, x2, y2, nRows = grid.size(), nCols = grid[0].size();
+    // Spiral filling of the open space
+    // Copy incoming list to 'end'
+    std::list<gridNode_t> pathNodes(init);
+    // Create iterator for gridNode_t list and let it point to the last element of end
+    std::list<gridNode_t>::iterator it = --(pathNodes.end());
+    if (pathNodes.size() > 1) // if list is length 1, keep iterator at end
+      it--;                   // Let iterator point to second to last element
+
+    gridNode_t prev = *(it);
+    bool done = false;
+    while (!done)
     {
-      x2 = pathNodes.back().pos.x + dx;
-      y2 = pathNodes.back().pos.y + dy;
-      if (x2 >= 0 && x2 < nCols && y2 >= 0 && y2 < nRows)
+      if (it != pathNodes.begin())
+      {
+        // turn ccw
+        dx = pathNodes.back().pos.x - prev.pos.x;
+        dy = pathNodes.back().pos.y - prev.pos.y;
+        dx_prev = dx;
+        dx = -dy;
+        dy = dx_prev;
+      }
+      else
       {
-        if (grid[y2][x2] == eNodeOpen && visited[y2][x2] == eNodeOpen)
+        // Initialize spiral direction towards y-axis
+        dx = 0;
+        dy = 1;
+      }
+      done = true;
```

Comment:
This is strange, could use a comment that it might be overwritten to false later.